src/ng/parse.js

```diff
@@ -1884,28 +1884,37 @@ function $ParseProvider() {
     function oneTimeWatchDelegate(scope, listener, objectEquality, parsedExpression, prettyPrintExpression) {
       var isDone = parsedExpression.literal ? isAllDefined : isDefined;
       var unwatch, lastValue;
-      if (parsedExpression.inputs) {
-        unwatch = inputsWatchDelegate(scope, oneTimeListener, objectEquality, parsedExpression, prettyPrintExpression);
-      } else {
-        unwatch = scope.$watch(oneTimeWatch, oneTimeListener, objectEquality);
-      }
+
+      var exp = parsedExpression.$$intercepted || parsedExpression;
+      var post = parsedExpression.$$interceptor || identity;
+
+      var useInputs = parsedExpression.inputs && !exp.inputs;
```

Comment:
It's basically detecting [this case](https://github.com/angular/angular.js/blob/v1.6.5/src/ng/parse.js#L1973). This is when the intercepted expression had no inputs, so the expression itself becomes the input.

For example
* `$parse("x", stringify)`: `x` has no inputs, but we don't want to execute `stringify(x)` per-digest so we set `x` itself as the input and `useInputs = true` to handle this
* `$parse("x | number", stringify)`: `x | number` already has `x` as an input so we just continue using that input (and `useInputs = false`)